crates/l1tx/src/filter/withdrawal_fulfillment.rs

```diff
@@ -0,0 +1,254 @@
+use bitcoin::{ScriptBuf, Transaction};
+use strata_primitives::l1::{BitcoinAmount, WithdrawalFulfillmentInfo};
+
+use super::TxFilterConfig;
+use crate::utils::op_return_nonce;
+
+fn create_opreturn_metadata(operator_idx: u32, deposit_idx: u32) -> ScriptBuf {
+    let mut metadata = [0u8; 8];
+    // first 4 bytes = operator idx
+    metadata[..4].copy_from_slice(&operator_idx.to_be_bytes());
+    // next 4 bytes = deposit idx
+    metadata[4..].copy_from_slice(&deposit_idx.to_be_bytes());
+    op_return_nonce(&metadata)
+}
+
+/// Parse transaction and search for a Withdrawal Fulfillment transaction to an expected address.
+pub fn parse_withdrawal_fulfillment_transactions<'a>(
+    tx: &'a Transaction,
+    filter_conf: &'a TxFilterConfig,
+) -> Option<WithdrawalFulfillmentInfo> {
+    // 1. Check this is of correct structure
+    let frontpayment_txout = tx.output.first()?;
+    let metadata_txout = tx.output.get(1)?;
+    if !metadata_txout.script_pubkey.is_op_return() {
+        return None;
+    }
+
+    // 2. Check withdrawal is to an address we expect
+    let withdrawal = filter_conf
+        .expected_withdrawal_fulfillments
+        .binary_search_by_key(&frontpayment_txout.script_pubkey, |expected| {
```

Comment:
The way this currently works would break if there's two withdrawals with the same dest spk.  We have to include the deposit idx in the op_return payload anyways, so we should be doing the lookup based on that.

Reply:
We have a check for the deposit idx and operator idx just after that, so this should be fine and just serves as redundant check.